test/api/test_ut_run.pkb

```diff
@@ -114,5 +114,107 @@ Failures:%
     execute immediate 'drop package test_stateful';
   end;
 
+  procedure run_in_invalid_state is
+    l_results   ut3.ut_varchar2_list;
+    l_actual    clob;
+    l_expected  varchar2(32767);
+  begin
+    select *
+      bulk collect into l_results
+    from table(ut3.ut.run('failing_invalid_spec'));
+    
+    l_actual := ut3.ut_utils.table_to_clob(l_results);
+    ut.expect(l_actual).to_be_like('%Call params for % are not valid: package does not exist or is invalid: %FAILING_INVALID_SPEC%'); 
+    
+  end;
+
+  procedure compile_invalid_package is
+    ex_compilation_error exception;
+    pragma exception_init(ex_compilation_error,-24344);
+    pragma autonomous_transaction;
+  begin
+    begin
+      execute immediate q'[create or replace package failing_invalid_spec as
+  --%suite
+  gv_glob_val non_existing_table.id%type := 0;
+
+  --%test
+  procedure test1;
+end;]';
+    exception when ex_compilation_error then null;
+    end;
+    begin
+      execute immediate q'[create or replace package body failing_invalid_spec as
+  procedure test1 is begin ut.expect(1).to_equal(1); end;
+end;]';
+    exception when ex_compilation_error then null;
+    end;
+  end;
+  procedure drop_invalid_package is
+    pragma autonomous_transaction;
+  begin
+    execute immediate 'drop package failing_invalid_spec';
+  end;
+
+  procedure run_and_revalidate_specs is
+    l_results   ut3.ut_varchar2_list;
+    l_actual    clob;
+    l_is_invalid number;
+  begin
+    execute immediate q'[select count(1) from all_objects o where o.owner = :object_owner and o.object_type = 'PACKAGE'
+            and o.status = 'INVALID' and o.object_name= :object_name]' into l_is_invalid
+            using user,'FAILING_INVALID_SPEC';
```

Comment:
It was tricky to understand what test is doing because the name used in this test `FAILING_INVALID_SPEC` is quite misleading.
Maybe change the name to something more like: `invalid_package_that_revalidates`?

Reply:
done